pkg/reconciler/event.go

```diff
@@ -63,3 +65,10 @@ func EmitEvent(c record.EventRecorder, beforeCondition *apis.Condition, afterCon
 		}
 	}
 }
+
+// EmitErrorEvent emits a failure associated to an error
+func EmitErrorEvent(c record.EventRecorder, err error, object runtime.Object) {
+	if err != nil {
```

Comment:
feels a bit odd to have this function take an error as an argument vs checking the error in the place where it originates - is this a pattern we do other places as well?

Reply:
Well, it's a bit different because in this case the error is the event that we want to send.

On knative they went a bit further, they have now an event object that also implements the error interface.
They have a generated `Reconcile` that invokes a `ReconcileKind` which returns an event that may be an error - at least that's how I understood it - so this goes a bit in that direction. 

I could check if the `err` is nil at the source, but it felt safer to do here.

Reply:
https://github.com/knative/pkg/tree/master/injection#generated-reconciler-responsibilities

Reply:
My plan for upcoming patches was to centralize the event logic in `event.go` i.e. to receive all requests events / error notifications from the taskrun/pipelinerun controller in there, and to decide there when to emit what, k8s event and/or cloudevent.